lib/LangInterface.php

```diff
@@ -993,6 +993,49 @@ public static function translate_current_url( string $lang ) : string {
 			}
 		}
 
+		// Check for user defined routes.
+		if ( $wp_the_query instanceof WP_Query ) {
+
+			/**
+			 * Filter to add translatable routes outside of posts, taxonomies and archives.
+			 *
+			 * Only the top route needs to be added, the same that was added to the rewrite rules.
+			 * For example: A route for accounts 'account' might have sub routes like
+			 * 'account/address'. Only the top route 'account' needs to be added to this filter,
+			 * since WordPress will match the top route in the rewrite rules.
+			 *
+			 * @since 0.3.2
+			 *
+			 * @param array  $routes Translatable routes.
+			 * @param string $lang   Language code to translate route to.
+			 * @return array
+			 */
+			$routes = apply_filters( 'ubb_translatable_routes', [], $lang );
+
+			// Try to match all the routes with the route that WordPress matched. First one matched is chosen.
+			foreach ( $routes as $route ) {
+
+				// If the route is matched, return the translated url.
+				if ( isset( $wp_the_query->query[ $route ] ) ) {
+
+					// Get base url for the language requested.
+					add_filter( 'ubb_current_lang', $fn = fn () => $lang );
```

Comment:
Why are we giving a function as the filtered value?

Reply:
The function is not the filtered value but the function that will be called by the filter to change its value.

Reply:
Ok, but then it could just be the function itself:
```suggestion
					add_filter( 'ubb_current_lang', fn () => $lang );
```

Reply:
I need the assigned variable in order to remove it right after in the `remove_filter`